### PR TITLE
Add link to semantic re-ranking in model ref

### PR DESCRIPTION
**Pull request by leemthompo**
making sure links are bidirectional with the [the semantic reranking docs page](https://www.elastic.co/guide/en/elasticsearch/reference/current/semantic-reranking.html#semantic-reranking-in-es)

### CODE REVIEW

**Review by davidkyle — DISMISSED** [review]
LGTM

File: docs/en/stack/ml/nlp/ml-nlp-model-ref.asciidoc [hunk 1]
```diff
@@ -148,6 +148,9 @@ Using `DPREncoderWrapper`:
 [discrete]
 [[ml-nlp-model-ref-text-similarity]]
 == Third party text similarity models
+
+You can use these text similarity models models for {ref}/semantic-reranking.html#semantic-reranking-in-es[semantic re-ranking].
```
Comment (marciw) [inline docs/en/stack/ml/nlp/ml-nlp-model-ref.asciidoc:152]:
```suggestion
You can use these text similarity models for {ref}/semantic-reranking.html#semantic-reranking-in-es[semantic re-ranking].
```

**Review by marciw — COMMENTED** [review]
Was just wandering by and noticed a duplicate word

**Review by szabosteve — APPROVED** [review]
LGTM, thank you!

**Pull request merged**